external/vulkancts/modules/vulkan/conditional_rendering/vktConditionalDrawTests.cpp

```diff
@@ -0,0 +1,454 @@
+	}
+
+	if (m_conditionalData.conditionEnabled)
+	{
+		m_vk.cmdEndConditionalRenderingEXT(*m_cmdBuffer);
+	}
+
+	endRenderPass(m_vk, *m_cmdBuffer);
+	endCommandBuffer(m_vk, *m_cmdBuffer);
+
+	submitCommandsAndWait(m_vk, device, queue, m_cmdBuffer.get());
+
+	// Validation
+	tcu::Texture2D referenceFrame(vk::mapVkFormat(m_colorAttachmentFormat), (int)(0.5 + WIDTH), (int)(0.5 + HEIGHT));
+								  referenceFrame.allocLevel(0);
+
+	const deInt32 frameWidth	= referenceFrame.getWidth();
+	const deInt32 frameHeight	= referenceFrame.getHeight();
+
+	const tcu::Vec4 clearColor = tcu::RGBA::black().toVec();
+	const tcu::Vec4 drawColor  = tcu::RGBA::blue().toVec();
+
+	tcu::clear(referenceFrame.getLevel(0), clearColor);
+
+	const tcu::Vec4 referenceColor = m_conditionalData.expectCommandExecution ?
+										drawColor : clearColor;
+
+	Draw::ReferenceImageCoordinates refCoords;
+
+	for (int y = 0; y < frameHeight; y++)
+	{
+		const float yCoord = (float)(y / (0.5*frameHeight)) - 1.0f;
+
+		for (int x = 0; x < frameWidth; x++)
+		{
+			const float xCoord = (float)(x / (0.5*frameWidth)) - 1.0f;
+
+			if ((yCoord >= refCoords.bottom &&
+				 yCoord <= refCoords.top	&&
+				 xCoord >= refCoords.left	&&
+				 xCoord <= refCoords.right))
+				referenceFrame.getLevel(0).setPixel(referenceColor, x, y);
+		}
+	}
+
+	const vk::VkOffset3D zeroOffset = { 0, 0, 0 };
+	const tcu::ConstPixelBufferAccess renderedFrame = m_colorTargetImage->readSurface(queue, m_context.getDefaultAllocator(),
+		vk::VK_IMAGE_LAYOUT_GENERAL, zeroOffset, WIDTH, HEIGHT, vk::VK_IMAGE_ASPECT_COLOR_BIT);
+
+	qpTestResult res = QP_TEST_RESULT_PASS;
+
+	if (!tcu::fuzzyCompare(log, "Result", "Image comparison result",
+		referenceFrame.getLevel(0), renderedFrame, 0.05f,
+		tcu::COMPARE_LOG_RESULT)) {
+		res = QP_TEST_RESULT_FAIL;
+	}
+
+	return tcu::TestStatus(res, qpGetTestResultName(res));
+};
+
+}	// anonymous
+
+ConditionalDrawTests::ConditionalDrawTests (tcu::TestContext &testCtx)
+	: TestCaseGroup	(testCtx, "draw", "Conditional Rendering Of Draw Commands")
+{
+	/* Left blank on purpose */
+}
+
+ConditionalDrawTests::~ConditionalDrawTests (void) {}
+
+void ConditionalDrawTests::init (void)
+{
+	for (int conditionNdx = 0; conditionNdx < DE_LENGTH_OF_ARRAY(Conditional::s_testsData); conditionNdx++)
+	{
+		const ConditionalData& conditionData = Conditional::s_testsData[conditionNdx];
+
+		tcu::TestCaseGroup* conditionalDrawRootGroup = new tcu::TestCaseGroup(m_testCtx, de::toString(conditionData).c_str(), "Conditionaly execute draw calls");
+
+		for (deUint32 commandTypeIdx = 0; commandTypeIdx < DRAW_COMMAND_TYPE_DRAW_LAST; ++commandTypeIdx)
+		{
+			const DrawCommandType	command	= DrawCommandType(commandTypeIdx);
+
+			ConditionalTestSpec testSpec;
+			testSpec.command = command;
+			testSpec.drawCalls = 4;
+			testSpec.conditionalData = conditionData;
+			testSpec.shaders[glu::SHADERTYPE_VERTEX] = "vulkan/conditional_rendering/VertexFetch.vert";
```

Comment:
I don't have a good answer besides writing these tests on different weeks. Will embed the shaders.

Reply:
Upon the second look I see why. I'm inheriting Draw::DrawTestsBaseClass since it has all boilerplate needed for testing draw functions and it is used with Draw::InstanceFactory - together they require shaders to be in separate files.

I don't have strong feeling towards embedding or not the shaders, I just tried to be consistent with other code: draw tests look like other draw tests, dispatch tests look like other dispatch tests.
Considering this, is still better to embed them in my draw tests?

Reply:
Either embed or reuse external/vulkancts/data/vulkan/dynamic_state/VertexFetch.*. I think you shaders are exactly the same (and actually not placed in the correct folder)

Reply:
Indeed, I messed up shaders placement. Now I'm reusing the ones from dynamic_state